frodo.js

```diff
@@ -3,15 +3,14 @@
  *
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?